### PR TITLE
fix: Limit the number of players that can join a team

### PR DESCRIPTION
**Pull request by aronwk-aaron**
after being invited

Tested that the server compiles and does not crash

### CODE REVIEW

**Review by EmosewaMC — APPROVED** [review]

**Review by Wincent01 — COMMENTED** [review]
Is there a way to notify the affected parties that this has happened? If not through something else, a chat message would probably be good.

**Comment by aronwk-aaron** [comment]
> Is there a way to notify the affected parties that this has happened? If not through something else, a chat message would probably be good.

Done

File: dChatServer/PlayerContainer.cpp [hunk 1]
```diff
@@ -209,6 +210,8 @@ TeamData* PlayerContainer::GetTeam(LWOOBJID playerID) {
 void PlayerContainer::AddMember(TeamData* team, LWOOBJID playerID) {
 	if (team->memberIDs.size() >= 4){
 		Game::logger->Log("PlayerContainer", "Tried to add player to team that already had 4 players");
+		auto* player = GetPlayerData(playerID);
+		ChatPackets::SendSystemMessage(player->sysAddr, u"The teams is full! You have not been added to a team!");
```
Comment (EmosewaMC) [inline dChatServer/PlayerContainer.cpp:214]: Check your pointers.

**Review by EmosewaMC — CHANGES_REQUESTED** [review]

File: dChatServer/PlayerContainer.cpp (hunk 1 above)
Comment (aronwk-aaron) [reply dChatServer/PlayerContainer.cpp:214]: tru

**Review by EmosewaMC — APPROVED** [review]

**Review by Jettford — APPROVED** [review]

**Pull request merged**